include/pmacc/math/vector/compile-time/Vector.hpp

```diff
@@ -182,11 +171,12 @@ namespace pmacc
                 typename T_TypeA1,
                 typename T_TypeB0,
                 typename T_TypeB1,
+                template<typename...>
                 typename T_BinaryOperator>
             struct applyOperator<CT::Vector<T_TypeA0, T_TypeA1>, CT::Vector<T_TypeB0, T_TypeB1>, T_BinaryOperator>
             {
-                using OpResult0 = typename mpl::apply<T_BinaryOperator, T_TypeA0, T_TypeB0>::type;
-                using OpResult1 = typename mpl::apply<T_BinaryOperator, T_TypeA1, T_TypeB1>::type;
+                using OpResult0 = T_BinaryOperator<T_TypeA0, T_TypeB0>;
```

Comment:
WARNING: This transition is changing maybe silently some definitions in PMAcc or PIConGPU.
`mpl::apply` is not assuming any order `X<boost::mpl_2,boost::mpl_1>` can be used which is not equal to `T_BinaryOperator<T_TypeA0, T_TypeB0>`
Could be that this is not important here but I need to check this part of code again.

Reply:
That depends on whether you consider `pmacc::math::CT::applyOperator` a public API that you want to use with MPL metafunctions. I saw that it was only used in the local file, so I refactored all uses of it to use the corresponding mp11 meta functions.

Your all, I can revert the code here.

Reply:
yes, let us keep it as it is.